### PR TITLE
METRON-822 Improve Fastcapa Performance

### PR DESCRIPTION
**Pull request by nickwallen**
This PR contains significant improvements to the performance and scalability of Fastcapa.  

* Previously the 'distributor' framework was used.  This did not scale well and has been replaced with a burst-oriented design.
* Receive and transmission functions have been separated to allow each to scale independently.
* Additional parameters have been added to allow the process to be tuned easily.
* Output provides basic transparency into the current state of processing.
* If the probe is overwhelmed with more packets than can be handled, it will continue processing the packets that it can without crashing.  
* The fact that packets are being dropped is very clear.
* A great deal of documentation has been added to the README.
* Support for latest release of DPDK; v16.11.1
* Support for latest release of RdKafka; v0.9.4

This change has been tested on Cisco UCS hardware with a 10G Cisco VNIC.  The probe was able to capture 1 gbps before packets started to drop.  Additional performance tuning would push this ceiling much higher, but for my purposes, I just needed to reach 1 gbps.  Additional work will proceed in the future to find its true performance ceiling.

To test the change yourself, simply spin-up the virtualized test environment which will deploy and validate that Fastcapa can land packets in Kafka correctly.
```
cd metron-deployment/vagrant/fastcapa-test-environment
vagrant up
```

### For all changes:
- [x] Is there a JIRA ticket associated with this PR?
- [x] Does your PR title start with METRON-XXXX?
- [x] Has your PR been rebased against the latest commit within the target branch?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root incubating-metron folder via:
- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### CODE REVIEW

**Comment by nickwallen** [comment]
The additional commits are showing up as Github is out-of-sync with Apache.  That should clear-up once they sync back up.

File: metron-sensors/fastcapa/src/args.c [hunk 1]
```diff
@@ -84,62 +95,167 @@ int parse_args(int argc, char** argv)
...
-            return -1;
-        }
+
+            // burst size
+            case 'b':
+                app.burst_size = atoi(optarg);
+                printf("[ -b BURST_SIZE ] defined as %d \n", app.burst_size);
+
+                if(app.burst_size < 1 || app.burst_size > MAX_BURST_SIZE) {
+                    printf("Invalid burst size; burst=%u must be in [1, %u]. \n", app.burst_size, MAX_BURST_SIZE);
```
Comment (cestella) [inline metron-sensors/fastcapa/src/args.c]: Can we send this to stderr?

Comment (cestella) [inline metron-sensors/fastcapa/src/args.c]: Actually, same comment on the error comments below.

File: metron-sensors/fastcapa/src/kafka.c [hunk 2]
```diff
@@ -21,11 +21,113 @@
...
+static int kaf_stats_cb(rd_kafka_t *rk, char *json, size_t UNUSED(json_len), void *opaque) 
+{
+    int rc;
+    struct opaque *data = (struct opaque*) opaque;
+    int conn_id = data->conn_id;   
+
+    // update queue depth of this kafka connection
+    kaf_conn_stats[conn_id].depth = rd_kafka_outq_len(rk);
+
+    // TODO this should be handled by a logging lib that can handle faults and rolling the output file
```
Comment (nickwallen) [inline metron-sensors/fastcapa/src/kafka.c]: Sure

File: metron-sensors/fastcapa/src/args.c (hunk 1 above)
Comment (nickwallen) [inline metron-sensors/fastcapa/src/args.c]: Sure, makes sense.

File: metron-sensors/fastcapa/src/kafka.c (hunk 2 above)
Comment (nickwallen) [inline metron-sensors/fastcapa/src/kafka.c]: Created [METRON-828](https://issues.apache.org/jira/browse/METRON-828) to track this.

**Comment by nickwallen** [comment]
As a comment on the JIRA itself, I added some details of a performance test showing the probe operating successfully at ~1.1 Gbps.

**Pull request closed without merging**